packages/integration/src/azure/config.ts

```diff
@@ -38,6 +38,75 @@ export type AzureIntegrationConfig = {
    * If no token is specified, anonymous access is used.
    */
   token?: string;
+
+  /**
+   * The credential to use for requests.
+   *
+   * If no credential is specified anonymous access is used.
+   */
+  credential?: Partial<AzureCredential>;
```

Comment:
Shouldn't this be something like this?

```suggestion
  credential?: AzureCredential;
```

Also I wonder if we should actually just change this to `authentication` and move `{ token?: string }` to be part of the union and deprecate the top level one? @backstage/maintainers?

Reply:
You are right, I did this because I used the type predicates inside `isValidCredential(...)` so it turns out typescript was not able to narrow the type to `ClientSecret | ManagedIdentity` after this `if`. Fixed it 👍 

I thought about making `token` part of the union but I looked at the `github` integration and there it's `token` or `apps` and this felt sort of similar. It would make the code simpler because then the type narrowing can replace some of the checks I added now.